lib/travis/logs/database.rb

```diff
@@ -92,6 +93,13 @@ def log_id_for_job_id(job_id)
         log[:id] if log
       end
 
+      def cached_log_id_for_job_id(job_id)
+        cache_key = "log_id.#{job_id}"
+        log_id = cache.read(cache_key) || log_id_for_job_id(job_id)
+        cache.write(cache_key, log_id) if log_id
```

Comment:
if the read was coming from the cache i believe you could spare this write here?